diff --git a/texk/web2c/Makefile.in b/texk/web2c/Makefile.in
index 2ebe8987..c9d4459f 100644
--- a/texk/web2c/Makefile.in
+++ b/texk/web2c/Makefile.in
@@ -2404,6 +2404,8 @@ am__depfiles_remade = ./$(DEPDIR)/aleph-aleph-pool.Po \
 am__mv = mv -f
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
 	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
+BUILD_COMPILE = @BUILD_CC@ $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
+	$(CPPFLAGS)$(CFLAGS)	
 LTCOMPILE = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) \
 	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
@@ -2416,6 +2418,9 @@ CCLD = $(CC)
 LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(AM_LDFLAGS) $(LDFLAGS) -o $@
+BUILD_CCLD = @BUILD_CC@
+BUILD_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC --mode=link \
+	$(BUILD_CCLD) -o $@
 AM_V_CCLD = $(am__v_CCLD_@AM_V@)
 am__v_CCLD_ = $(am__v_CCLD_@AM_DEFAULT_V@)
 am__v_CCLD_0 = @echo "  CCLD    " $@;
@@ -8133,7 +8138,8 @@ gftype$(EXEEXT): $(gftype_OBJECTS) $(gftype_DEPENDENCIES) $(EXTRA_gftype_DEPENDE
 
 himktables$(EXEEXT): $(himktables_OBJECTS) $(himktables_DEPENDENCIES) $(EXTRA_himktables_DEPENDENCIES) 
 	@rm -f himktables$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(himktables_OBJECTS) $(himktables_LDADD) $(LIBS)
+	# $(AM_V_CCLD)$(LINK) $(himktables_OBJECTS) $(himktables_LDADD) $(LIBS)
+	$(AM_V_CCLD)$(BUILD_LINK) $(himktables_OBJECTS) $(himktables_LDADD) $(LIBS)
 hitexdir/$(am__dirstamp):
 	@$(MKDIR_P) hitexdir
 	@: > hitexdir/$(am__dirstamp)
@@ -9266,7 +9272,11 @@ am--depfiles: $(am__depfiles_remade)
 @am__fastdepCC_TRUE@	$(am__mv) $$depbase.Tpo $$depbase.Po
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='$<' object='$@' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(COMPILE) -c -o $@ $<
+	if test "$@" = "himktables.o"; then \
+		$(AM_V_CC)$(BUILD_COMPILE) -c -o $@ $< \
+	else \
+	@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(COMPILE) -c -o $@ $< \
+	fi
 
 .c.obj:
 @am__fastdepCC_TRUE@	$(AM_V_CC)depbase=`echo $@ | sed 's|[^/]*$$|$(DEPDIR)/&|;s|\.obj$$||'`;\
diff --git a/texk/web2c/configure b/texk/web2c/configure
index 682ae7fe..84501063 100755
--- a/texk/web2c/configure
+++ b/texk/web2c/configure
@@ -658,6 +658,7 @@ ac_subst_vars='am__EXEEXT_FALSE
 am__EXEEXT_TRUE
 LTLIBOBJS
 LIBOBJS
+BUILD_CC
 subdirs
 LIBSYNCTEX
 LTLIBSYNCTEX
@@ -7984,8 +7985,10 @@ fi
 
 test -z "$RANLIB" && RANLIB=:
 
-
-
+if test -n "$build_alias"; then
+  build_cc="gcc"
+  BUILD_CC=$build_cc
+fi
 
 
 
